Dockerfile.rhel7

```diff
@@ -0,0 +1,28 @@
+FROM rhel7:latest
+
+ENV GOPATH=/go \
+    PATH="${PATH}:${GOPATH}/bin"
+RUN mkdir $GOPATH
+
+COPY . $GOPATH/src/github.com/aerogear/aerogear-app-metrics
+
+RUN yum install -y golang make git \
```

Comment:
Thanks for the feedback & thorough review @darahayes!

> I wonder could we take advantage of multi-stage builds?

Yes! :) I even mentioned multi-stage builds as a future improvement in the relevant commit message (it didn't fit on the summary line though, so you probably didn't see it).

> However, the resulting image has lots of extra stuff that's potentially not needed. Like the source code, and all of the Golang tooling, etc.

For the RPMs that get installed here, they get removed on lines 13 & 14 (same RUN instruction, so they shouldn't exist in any layer in the outputted image). There's still everything that got copied into `$GOPATH`. I think the multi-stage build would be the ultimate solution to fixing that though (I've got a sneaking suspicion that removing them in a RUN instruction like this would actually make the final image larger (if it doesn't get squashed).

> Also, one other tangentially related question. Are we ensuring a specific Golang version here? The upstream project is tested against a specific Golang version and that's the same version that ends up in the upstream container image. Do we know what `yum install -y golang` will give us?

It'll be whatever the latest version of golang that is provided in the relevant rhel repo. At the moment, that appears to be `1.9.4`. Possibly related: I notice that in the [circleci config file](https://github.com/aerogear/aerogear-app-metrics/blob/master/.circleci/config.yml), there are references to both `golang:1.9` and `golang:1.10` in different sections. Is that intentional?